### PR TITLE
Try the mkdocs search hack

### PR DESCRIPTION
**Pull request by lwm**
Something for https://github.com/commercialhaskell/stack/issues/3376.

Unfortunately, I couldn't test this (because mkdocs search actually works locally). For someone who has access to the RTD builder, I think you can build off a branch and view the documentation. Then just try to search something and see if you get the mkdocs search bar and not the broken RTD one.

\cc @mgsloan @borsboom

### CODE REVIEW

File: etc/mkdocsjs/searchhack.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Please see https://github.com/commercialhaskell/stack/issues/3376
+
+(function () {
```
Comment (mgsloan) [inline etc/mkdocsjs/searchhack.js]: Haven't tested it, but it looks to me like this function won't be run, needs another () on the last line.

**Comment by borsboom** [comment]
@lwm I've copied your `fixup-search` branch to the main repo, and told readthedocs to build it.  Unfortunately, all documentation builds seem to be failing for the last day with "An unexpected error occurred", so we can't actually test it yet.  You should be able to push changes to the `fixup-search` branch yourself too.

File: etc/mkdocsjs/searchhack.js (hunk 1 above)
Comment (lwm) [inline etc/mkdocsjs/searchhack.js]: Ah yes, you're right. I should have listened to my JavaScript linter ;)

**Comment by lwm** [comment]
> all documentation builds seem to be failing for the last day with "An unexpected error occurred"

Hmm, seems like the RTD build needs some attention then. Any stack trace? 

Feel like adding me to the RTD account so I can play around with it or?

**Comment by borsboom** [comment]
RTD seems to be failing most builds, but occasionally passing.  I've made `lwm` a maintainer there.

**Comment by lwm** [comment]
Ugh, seems to be a bunch of these 'unexpected error' messages plaguing people, as far as I can see in https://github.com/rtfd/readthedocs.org/issues/2985. I'll keep an eye there but can't seem to get any good of it now.

**Comment by lwm** [comment]
I'll close this off for now and try to fix when things are clear on the RTD side.

**Pull request closed without merging**